From 1a5488f4d4cdf48dece1efac765dabc63e1e9c94 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 14:21:52 +0800
Subject: [PORT FROM MAIN] libstagefright: A fix to ensure we can get the closest sync frame to the seek point

BZ: 55341

JB adopts binary search for the closest sync frame
to the seek point. We need to have the left end
subtract 1 to meet the need.

Change-Id: I67eca217ceb4b38b1b6cd0f8b92d5316809aa28e
Orig-Change-Id: I209bf7986abed33439c2bab65dc33d1a46aa4019
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/SampleTable.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/SampleTable.cpp b/media/libstagefright/SampleTable.cpp
index 33422bd..75f63d5 100644
--- a/media/libstagefright/SampleTable.cpp
+++ b/media/libstagefright/SampleTable.cpp
@@ -645,6 +645,10 @@ status_t SampleTable::findSyncSampleNear(
         left = left - 1;
     }
 
+    if (left > 0) {
+        --left;
+    }
+
     // Now ssi[left] is the sync sample index just before (or at)
     // start_sample_index.
     // Also start_sample_index < ssi[left + 1], if left + 1 < mNumSyncSamples.
-- 
1.7.4.1

